---
title: "Tissue specificity index analysis - tau"
author: "Deisiany"
date: "2024-01-13"
output:
  pdf_document: default
  word_document: default
  html_document: default
---
# Load libraries
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
```

# read in data
For Tissue specificity index analysis - tau - we will use the 133 genes which had average expression across all tissues equal or greater than 2, and was used in our PCA analysis.

```{r}
# Load full genes
data <- read_csv("Tau_133_genes_with_expression_across_tissues_are_greater_than_2_SoyARCs.csv") 
data <- data %>% select(-c(Class, Clade, expr_median))
```

# Calculate the component value
X(^) = Xi/ max(Xi) for 1<=i <=n
```{r}
# lets start by assigning our data to df
max_component_df <- data

for (Xi in 1:nrow(max_component_df)) {
  max_component_df[Xi, 4:17] <- max_component_df[Xi, 4:17] / max(max_component_df[Xi, 4:17], na.rm = TRUE)
}

max_component_df

# Calculate Tau (Tissue Specificity Index) for each gene
tau_df <- max_component_df

tau_df$tau <- numeric(nrow(tau_df))  # Create an empty column

for (i in 1:nrow(tau_df)) {
  tau_df$tau[i] <- sum(1 - tau_df[i, 4:17]) / 13
}

# write_csv(tau_df, "tau_df_contains_maximal_component_values.csv")
```

```{r}
(tau_df_values <- tau_df %>% select(`Transcript ID`, heatmap_label, tau))

#write_csv(tau_df_values, "tau_df_values.csv")
```

# tau values greater than .80 are a cut off utilized in the literature to denote tissues specificity.

```{r}
tau_greater_80 <- tau_df_values[tau_df_values$tau >= 0.80, ]

#write_csv(tau_greater_80, "tau_greater_80.csv")
```

Here https://rdrr.io/github/roonysgalbi/tispec/f/vignettes/UserGuide.Rmd we can access some useful information about the tau measurement. And these papers also mention these same analysis: https://academic.oup.com/bioinformatics/article/21/5/650/220059#394030419 
https://academic.oup.com/bib/article/18/2/205/2562739?login=true 
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7791837/#CR75 

# tau distribution for all 14 tissues
```{r}
tau_df <- read_csv( "tau_df_contains_maximal_component_values.csv")


library(ggplot2)
library(ggridges)
library(dplyr)

# Calculate medians
medians <- tau_df %>% 
  group_by(Family) %>% 
  summarise(median_tau = median(tau))

ggplot(tau_df, aes(x = tau, y = Family, fill = Family)) +
  scale_fill_manual(values = c("#86C5D8", "#620093", "#E7C94C")) +
  ggridges::geom_density_ridges() +
  ggridges::stat_density_ridges(
    quantile_lines = TRUE, 
    quantiles = 2) +
  coord_cartesian(ylim = c(1, length(unique(tau_df$Family)) + 1))

```

# statistical testing
```{r}
stats_tau <- tau_df %>% select(Family, tau)

summary(stats_tau)


library(doBy)
summaryBy(tau ~ Family,
          data = stats_tau,
          FUN = median)

ggplot(stats_tau) +
  aes(x = Family, y = tau, fill = Family) +
  geom_boxplot() +
  theme(legend.position = "none") +
  scale_fill_manual(values = c("#86C5D8", "#620093", "#E7C94C")) +
  theme_bw()

```
The null and alternative hypotheses of the Kruskal-Wallis test are:

ùêª0: The 3 families are equal in terms of tau
ùêª1: At least one Family is different from the other 2 Families in terms of tau

# kruskal test 

```{r}
kruskal.test(tau ~ Family, data = stats_tau)
```
based on the kruskal test we reject the null hypothesis and conclude that at least one family is different in terms of tau values.

# Post-hoc test
```{r}
pairwise.wilcox.test(stats_tau$tau, stats_tau$Family,
  p.adjust.method = "holm"
)
```

all 3 families are differ significantly in terms of tau, p<0.05

```{r}
# install newest version of ggplot2 and load all packages
#library(remotes)
# remotes::install_version("ggplot2", version = "3.5.1")
#install.packages("ggstatsplot")
library(ggplot2)
library(ggstatsplot)

stats_tau$Family[stats_tau$Family == "AFB/TIR1"] <- "TIR1/AFB"

ggstatsplot::ggbetweenstats(
  data = stats_tau,
  x = Family,
  y = tau,
  type = "nonparametric", #  Kruskal-Wallis
  plot.type = "box",
  # centrality.type = "nonparametric", #if using median values
  pairwise.comparisons = TRUE,
  pairwise.display = "significant",
  centrality.plotting = FALSE, 
  bf.message = FALSE) +
  scale_color_manual(values = 
                      c("#86C5D8", "#620093", "#E7C94C"))




ggsave("Kruskal-test-tau-values.pdf", height = 5, width = 6)
ggsave("Kruskal-test-tau-values.jpeg", height = 5, width = 6, dpi = 1000)
```


# Repeat analysis and order according to p-value
```{r}
# Load necessary packages
library(dplyr)
library(rstatix)
library(ggstatsplot)

# Compute pairwise comparisons using Wilcoxon test
pairwise_results <- stats_tau %>%
  pairwise_wilcox_test(tau ~ Family, p.adjust.method = "BH")

# Order the families based on the p-values of pairwise comparisons
# Assuming you want to order by the most significant difference
significant_order <- pairwise_results %>%
  arrange(p) %>%
  select(group1, group2) %>%
  unique() %>%
  unlist() %>%
  unique()

# Reverse the order to ensure it goes from left to right
significant_order <- rev(significant_order)

# Convert Family to a factor with levels in the desired order
stats_tau <- stats_tau %>%
  mutate(Family = factor(Family, levels = significant_order))


ggstatsplot::ggbetweenstats(
  data = stats_tau,
  x = Family,
  y = tau,
  type = "nonparametric",          # Kruskal-Wallis
  plot.type = "box",
  centrality.plotting = FALSE,
  pairwise.comparisons = TRUE,
  pairwise.display = "significant",
  bf.message = FALSE
) +
  scale_color_manual(values = c("#86C5D8", "#620093", "#E7C94C"))


ggsave("Kruskal-test-tau-values.pdf", height = 5, width = 6)
ggsave("Kruskal-test-tau-values.jpeg", height = 5, width = 6, dpi = 1000)
```

# repeat analysis removing tissues that were not used in our final analysis
Roots, callus, Inflorescence and so on, were not taken into account in our final analysis. Thus, lets repeat this without this tissues.

```{r}
# lets start by assigning our data to df
max_component_df_aerial <- data %>% 
  select(`Transcript ID`, heatmap_label, SAM6D, SAM17D, SAM38D, AM, Cotyledon, Leaf, Hypocotyl)

for (Xi in 1:nrow(max_component_df_aerial)) {
  max_component_df_aerial[Xi, 3:9] <- max_component_df_aerial[Xi, 3:9] / max(max_component_df_aerial[Xi, 3:9], na.rm = TRUE)
}

max_component_df_aerial

# Calculate Tau (Tissue Specificity Index) for each gene
tau_df_aerial <- max_component_df_aerial

tau_df_aerial$tau <- numeric(nrow(tau_df_aerial))  # Create an empty column

for (i in 1:nrow(tau_df_aerial)) {
  tau_df_aerial$tau[i] <- sum(1 - tau_df_aerial[i, 3:9]) / 6
}

#write_csv(tau_df_aerial, "tau_df_values_aerial_tissues.csv")

tau_df_aerial_values <- tau_df_aerial %>% select(`Transcript ID`, heatmap_label, tau)

# Now lets select values greater than 80 again
tau_aerial_greater_80 <- 
  tau_df_aerial_values[tau_df_aerial_values$tau >= 0.80, ]

```




